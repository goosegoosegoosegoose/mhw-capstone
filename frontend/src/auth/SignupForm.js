import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import { Form, Button } from "react-bootstrap"

const SignupForm = ({signup}) => {
  const INITIAL_STATE = {
    username: "",
    password: "",
    email: ""
  };
  const [formData, setFormData] = useState(INITIAL_STATE);
  const nav = useNavigate();

  const handleChange = evt => {
    const { name, value } = evt.target;
    setFormData(fData => ({
      ...fData,
      [name]: value
    }))
  };

  const handleSubmit = evt => {
    evt.preventDefault();
    signup({...formData});
    setFormData(INITIAL_STATE);
    nav(-1);
  };

  return (
    <div className="container">
      <div className="my-2 mx-2">
        <h2 className="my-2">Sign Up</h2>
        <Form onSubmit={handleSubmit}>
          <Form.Group className="my-4">
            <Form.Label htmlFor="username">Username</Form.Label>
            <Form.Control type="text" name="username" value={formData.username} onChange={handleChange} />
          </Form.Group>
          <Form.Group className="my-4">  
            <Form.Label htmlFor="password">Password</Form.Label>
            <Form.Control type="text" name="password" value={formData.password} onChange={handleChange} />
          </Form.Group>
          <Form.Group className="my-4">
            <Form.Label htmlFor="email">Email</Form.Label>
            <Form.Control type="text" name="email" value={formData.email} onChange={handleChange} />
          </Form.Group>
          <Button type="submit" variant="primary">Submit</Button>
        </Form>
      </div>
    </div>
  )
}

export default SignupForm;