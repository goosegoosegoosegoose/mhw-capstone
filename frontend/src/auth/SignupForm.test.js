import { render } from '@testing-library/react';
import SignupForm from "./SignupForm";

test('smoke test', () => {
  render(      
    <SignupForm />
  );
});

test('snapshot', () => {
  const { asFragment } = render(<SignupForm />);
  expect(asFragment()).toMatchSnapshot();
});