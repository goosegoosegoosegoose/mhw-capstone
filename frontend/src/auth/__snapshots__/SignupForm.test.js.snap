// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="my-2 mx-2"
    >
      <h2
        class="my-2"
      >
        Sign Up
      </h2>
      <form
        class=""
      >
        <div
          class="my-4"
        >
          <label
            class="form-label"
            for="username"
          >
            Username
          </label>
          <input
            class="form-control"
            name="username"
            type="text"
            value=""
          />
        </div>
        <div
          class="my-4"
        >
          <label
            class="form-label"
            for="password"
          >
            Password
          </label>
          <input
            autocomplete="on"
            class="form-control"
            name="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="my-4"
        >
          <label
            class="form-label"
            for="email"
          >
            Email
          </label>
          <input
            class="form-control"
            name="email"
            type="text"
            value=""
          />
        </div>
        <button
          class="btn btn-primary"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
